compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/QueryExpressionTest.java

```diff
@@ -281,4 +281,58 @@ public void testQueryActionOrExprWithAsyncSendActionInJoinClause() {
     public void testQueryWithMapKeyword() {
         test("map from int a in b select e", "query-expr/query_expr_assert_79.json");
     }
+
+    @Test
+    public void testQueryWithGroupByUsingVarDefGroupingKeyWithVar() {
+        testFile("query-expr/query_expr_source_80.bal", "query-expr/query_expr_assert_80.json");
```

Comment:
Let's move valid syntax tests to the top

Reply:
Didn't quite understand what you have meant by this. The valid ones are already at the top, the invalid ones are all inside `testQueryWithGroupByRecovery()` after the valid scenarios.